### PR TITLE
Adding settings page

### LINKED ISSUE #37: Adding Settings Page
what needs to be done:

Adding a Settings page that allows for updating:
- Nickname
- First name
- Last name

### PR DESCRIPTION
**Pull request by koukouihssane**
Adding settings page

### CODE REVIEW

File: frontend/src/components/auth/2fa/otp/qrcode/ImageReverser.tsx [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+	const filterStyle = {
+		filter: "invert(100%)",
+	};
+
+	return (
+		<Image
+			width={250}
+			height={250}
+			src={imageUrl}
+			alt="Reversed Colors"
```
Comment (MouadxBth) [inline frontend/src/components/auth/2fa/otp/qrcode/ImageReverser.tsx:17]: Alt should have been taken in props

File: frontend/src/components/auth/2fa/otp/qrcode/ImageReverser.tsx
Comment (MouadxBth) [inline frontend/src/components/auth/2fa/otp/qrcode/ImageReverser.tsx:1]: the old style was not so bad, but okay

File: frontend/src/components/generic/GenericFormField.tsx [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+	type,
+	placeholder,
+	className,
+}: GenericFormFieldProps<ProfileFormType>) => {
+	return (
+		<FormField
+			control={control}
+			name={name}
+			render={({ field }) => (
+				<>
```
Comment (MouadxBth) [inline frontend/src/components/generic/GenericFormField.tsx]: Unnecessary React Fragment

Comment (koukouihssane) [reply frontend/src/components/generic/GenericFormField.tsx]: fixed

File: frontend/src/hooks/usProfileForm.tsx
Comment (MouadxBth) [inline frontend/src/hooks/usProfileForm.tsx:1]: File name is incorrect

**Review by MouadxBth — APPROVED** [review]
Reviewed Successfully!

**Pull request merged**